tests/run-pass/communication.rs

```diff
@@ -0,0 +1,6 @@
+// ignore-windows: TODO env var emulation stubbed out on Windows
+// compile-flags: -Zmiri-enable-communication
+
+fn main() {
+    assert_eq!(std::env::var("MIRI_ENV_VAR_TEST"), Ok("0".to_owned()));
```

Comment:
I like this test!

Please also add a test to `env.rs` asserting `std::env::var("MIRI_ENV_VAR_TEST").is_err()`, just to make sure it is indeed properly isolated.